example/tutorial_mnist_horovod.py

```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+import tensorflow as tf
+import tensorlayer as tl
+import numpy as np
+
+tf.logging.set_verbosity(tf.logging.DEBUG)
+tl.logging.set_verbosity(tl.logging.DEBUG)
+
+
+def make_dataset(images, labels):
+    ds1 = tf.data.Dataset.from_tensor_slices(images)
+    ds2 = tf.data.Dataset.from_tensor_slices(np.array(labels, dtype=np.int64))
+    return tf.data.Dataset.zip((ds1, ds2))
+
+
+def model_function(x, y_):
+    with tf.variable_scope('mlp', reuse=tf.AUTO_REUSE):
+        network = tl.layers.InputLayer(x, name='input')
+        network = tl.layers.DropoutLayer(network, keep=0.8, name='drop1', is_fix=True)
+        network = tl.layers.DenseLayer(network, 800, tf.nn.relu, name='relu1')
+        network = tl.layers.DropoutLayer(network, keep=0.5, name='drop2', is_fix=True)
+        network = tl.layers.DenseLayer(network, 800, tf.nn.relu, name='relu2')
+        network = tl.layers.DropoutLayer(network, keep=0.5, name='drop3', is_fix=True)
+        network = tl.layers.DenseLayer(network, n_units=10, act=tf.identity, name='output')
+        return network, tl.cost.cross_entropy(network.outputs, y_, name='cost')
+
+
+# load mnist data
+X_train, y_train, X_val, y_val, X_test, y_test = tl.files.load_mnist_dataset(shape=(-1, 784))
+dataset = make_dataset(X_train, y_train)
+
+optimizer = tf.train.RMSPropOptimizer(0.001)
+trainer = tl.distributed.HorovodTrainer(model_function, dataset, optimizer)
+trainer.run()
```

Comment:
while not trainer.sess.should_stop():
      trainer.train_batch(self)
      if trainer.is_master and (sess.run(global_step) % 10000 == 0):
            tl.files.save_npz(trainer.net, ‘mode.npz’)